src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
Note that here we are assigning an `uint32_t` to a `long`, not the other way. So, there's no problem.
The variable being assigned to (eap->argt) is of type long.
The value being assigned (uc->uc_argt) is a member of struct ucmd, of type uint32_t. I
Before refactoring (struct ucmd).uc_argt, I checked all values assigned to it never needed more than 4 bytes, and were always positive. Thus I refactored it into an uint32_t. Probably, the same will be true for (struct exarg).argt, but I haven't checked that yet.
This was one of many places were long (and long_u) are used when only 4 bytes are required (which had sense for 16bit platforms, but not for 32bit and above).

What it's true, it's that there's a missing cast here, to avoid warning for mixing signed/unsigned types, which is not being signaled because -Wconversion is not active on ex_docmd.c. The line should be:
```
eap->argt = (long)uc->uc_argt
```